app/src/main/java/com/vmware/nimbus/ui/main/OptionsActivity.java

```diff
@@ -47,6 +55,16 @@ public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
             }
         });
 
+        loginSource.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
+                settings.edit().putBoolean(
+                        getApplicationContext().getResources().getString(R.string.login_source_property_name),
+                        isChecked
```

Comment:
Changing the login source property and then navigating back to the login page does not change the login method shown because it goes to a previous state, not using "onCreate" in CspLoginActivity. Need a way to navigate to the new login page.